From: "weiyongjun (A)" <weiyongjun1@huawei.com>
Date: Wed, 17 Jan 2018 11:28:38 +0000
Subject: RDMA/hns: Remove unnecessary platform_get_resource() error check
Patch-mainline: v4.16-rc1
Git-commit: 0b5fe5c43ab67130d259d800fcc109c7340acb63
References: bsc#1104427 FATE#326416

devm_ioremap_resource() already checks if the resource is NULL, so
remove the unnecessary platform_get_resource() error check.

Signed-off-by: Wei Yongjun <weiyongjun1@huawei.com>
Reviewed-by: Leon Romanovsky <leonro@mellanox.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/hns/hns_roce_hw_v1.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/infiniband/hw/hns/hns_roce_hw_v1.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v1.c
@@ -4788,10 +4788,6 @@ static int hns_roce_get_cfg(struct hns_r
 
 	/* get the mapped register base address */
 	res = platform_get_resource(hr_dev->pdev, IORESOURCE_MEM, 0);
-	if (!res) {
-		dev_err(dev, "memory resource not found!\n");
-		return -EINVAL;
-	}
 	hr_dev->reg_base = devm_ioremap_resource(dev, res);
 	if (IS_ERR(hr_dev->reg_base))
 		return PTR_ERR(hr_dev->reg_base);
